### PR TITLE
PWN-5027 - make amounts view works as expected + limits and max amount working also

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5027

## Description of Work

Make amounts view works as expected + limits and max amount working also

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/solend/model/SolendDepositToken.kt [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
+    open val usdRate: BigDecimal,
     open val iconUrl: String?,
-    open val supplyInterest: BigDecimal?
+    open val supplyInterest: BigDecimal?,
 ) : Parcelable {
 
+    var availableTokensForDeposit: BigDecimal? = null
+
     @Parcelize
     data class Active constructor(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/solend/model/SolendDepositToken.kt:18]: no need to constructor here

File: app/src/main/java/org/p2p/wallet/solend/ui/deposit/SolendDepositFragment.kt [hunk 2]
```diff
@@ -115,6 +128,116 @@ class SolendDepositFragment :
...
+        viewDoubleInput.setBottomMessageText(
+            getString(
+                R.string.solend_deposit_bottom_message_with_amount,
+                input.scaleLong(), output.scaleShort()
+            )
+        )
+        buttonInfo.isVisible = true
+        buttonInfo.apply {
+            setIconResource(R.drawable.ic_info_outline)
+            iconTint = ContextCompat.getColorStateList(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/solend/ui/deposit/SolendDepositFragment.kt:224]: i guess we need a new extension for the Context / Button...

File: app/src/main/java/org/p2p/wallet/solend/ui/deposit/SolendDepositPresenter.kt [hunk 3]
```diff
@@ -30,14 +28,12 @@ class SolendDepositPresenter(
...
-                    validDeposits = solendDepositsInteractor.getUserDeposits()
-                        .filter { depositToken ->
-                            userInteractor.getUserTokens()
-                                .firstOrNull { token -> token.tokenSymbol == depositToken.tokenSymbol }
-                                ?.totalInUsd
-                                .orZero()
-                                .isNotZero()
-                        }
+                    val userTokens = userInteractor.getUserTokens()
+                    validDeposits = solendDepositsInteractor.getUserDeposits().onEach { depositToken ->
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/solend/ui/deposit/SolendDepositPresenter.kt:32]: please revert the formatting, it's unreadable :(

File: app/src/main/java/org/p2p/wallet/solend/ui/deposit/SolendDepositPresenter.kt [hunk 4]
```diff
@@ -30,14 +28,12 @@ class SolendDepositPresenter(
...
+                        depositToken.availableTokensForDeposit = userTokens.firstOrNull { token ->
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/solend/ui/deposit/SolendDepositPresenter.kt:33]: i suggest to do copy() instead of var

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/solend/ui/deposit/SolendDepositPresenter.kt:33]: and maybe fill availableTokensForDeposit on the interactor level or repository level

if the presenter and domain requires availableTokensForDeposit to be filled, it's the repository responsibility to do that in the mapper

Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/solend/ui/deposit/SolendDepositPresenter.kt:33]: we need to move it up on root of feature level

**Review by 1sakov — APPROVED** [review]

**Pull request merged**